### PR TITLE
US 30535 - Terraform - FSLogix with ADDS v2 #429

### PR DESCRIPTION
**Pull request by sihbher**
# Overview/Summary

Adds the Management VM module to Terraform, including joining the storage account to the domain and a user assigned managed identity.

## This PR fixes/adds/changes/removes

1. Adds Mgmt VM module to Terraform workload.
2. Adds some variables used for Mgmt VM

## Testing Evidence

image

image

## As part of this Pull Request I have

- [ ] Read the [Contribution Guide](https://github.com/Azure/avdaccelerator/blob/main/CONTRIBUTING.md) and ensured this PR is compliant with the guide
- [ ] Ensured the resource API versions in `.bicep` file/s I am adding/editing are using the latest API version possible
- [ ] Checked for duplicate [Pull Requests](https://github.com/Azure/avdaccelerator/pulls)
- [ ] Associated it with relevant [GitHub Issues](https://github.com/Azure/avdaccelerator/issues)
- [ ] *(AVD LZA Team Only)* Associated it with relevant [ADO Items](https://dev.azure.com/CSUSolEng/Accelerator%20-%20AVD/_backlogs/backlog/Accelerator%20-%20AVD%20Team/Features)
- [ ] Ensured my code/branch is up-to-date with the latest changes in the `main` [branch](https://github.com/Azure/avdaccelerator/tree/main)
- [ ] Performed testing and provided evidence.
- [ ] Updated relevant and associated documentation (e.g. Contribution Guide, Module READMEs, Docs etc.)

### CODE REVIEW

**Comment by sihbher** [comment]
Talked to @danycontre and I added a feature to run custom script "Set-SessionHostConfiguration.ps1"  in order to configure fslogix on for every host.

image

image

**Comment by jensheerin** [comment]
do not merge until review is completed.

File: workload/terraform/greenfield/ADDSscenario/provider.tf [hunk 1]
```diff
@@ -14,13 +16,18 @@ terraform {
...
+      version = "=1.8.0"
     }
     time = {
       source = "hashicorp/time"
     }
   }
 }
 
+provider "azuread" {
+  tenant_id = "b7b812fe-be8c-4adc-883f-2a3c36d753d7"
```
Comment (jensheerin) [inline workload/terraform/greenfield/ADDSscenario/provider.tf:28]: The tenant_id needs to be removed, it is not needed here.

File: workload/terraform/greenfield/ADDSscenario/terraform.tfvars.sample [hunk 2]
```diff
@@ -39,6 +39,11 @@ hub_subscription_id       = "#######-####-####-acbc-##########"   # subscription
 identity_subscription_id  = "#######-####-####-acbc-##########"   # subscription where identity resources are deployed (If you are using a single subscription for hub and spoke, use the same subscription id as spoke_subscription_id)
 avdshared_subscription_id = "#######-####-####-acbc-##########"   # optional subscription where shared resources are deployed (If you are using a single subscription for hub and spoke, use the same subscription id as spoke_subscription_id)
 
+use_gallery_image = false                                           # set to true if you want to use a gallery image for the management VM
+ou_for_fslogix_storage = "OU=AvdStorage,DC=avd,DC=local"            # If you want to use a different OU for FSLogix storage     
+ou_name_for_fslogix_storage = "AvdStorage"                          # Name of the OU for FSLogix storage
+
+
 ############## Do not modify below this line unless you want to change naming convention#####################
 # Resource Groups
```
Comment (jensheerin) [inline workload/terraform/greenfield/ADDSscenario/terraform.tfvars.sample:48]: line 16 should be ou_for_session_hosts

**Comment by jensheerin** [comment]
also run fmt

File: workload/terraform/greenfield/ADDSscenario/provider.tf (hunk 1 above)
Comment (sihbher) [reply workload/terraform/greenfield/ADDSscenario/provider.tf:28]: Removed

File: workload/terraform/greenfield/ADDSscenario/terraform.tfvars.sample (hunk 2 above)
Comment (sihbher) [reply workload/terraform/greenfield/ADDSscenario/terraform.tfvars.sample:48]: Adjusted

**Comment by sihbher** [comment]
@jensheerin I've done the requested adjustments, please review

**Pull request closed without merging**